lib/BridgedRoom.js

```diff
@@ -434,6 +453,98 @@ BridgedRoom.prototype.onGitterMessage = function(message) {
+    await this._listGitterUsers();
+
+    if (!this._gitterAdminList.has(fromUser._gitter_id)) {
+        await this._main.getRelayGhost().send(this._gitterRoomName, "Only admins may use the !matrix command");
+        return;
+    }
+
+
+    const commandParts = message.substr("!matrix ".length).split(" ");
+    const command = commandParts[0].toLowerCase();
+    const commandArgs = commandParts.slice(1);
+
+    const roomIds = this.getAllMatrixRoomIds();
+
+    let response = "Command not understood";
+    if ((command === "kick" || command === "ban") && commandArgs.length === 0) {
+        response = `\`${command}\` needs a name parameter. Specify a matrix user ID (@username:domain) or displayname`;
+    } else if (command === "kick" || command === "ban") {
+        const givenUser = commandArgs[0];
+        let userId;
+        const isUserId = IS_USERID.exec(givenUser) !== null;
+        if (isUserId) {
+            userId = givenUser;
+            log.debug(`Matching '${givenUser}' as user id`);
+        }
+        // Because we can't ask the gitter admin to target a single room, we run this across
+        // all rooms. If we manage to do it once, consider it a success.
+        let actioned = false;
+        let foundUser = false;
+        for (const roomId of roomIds) {
+            let joinedMembers;
+            try {
+                joinedMembers = await this._main.getJoinedMembers(roomId);
+            } catch (ex) {
+                log.warn(`Request to joined_members failed in ${roomId}. Ignoring room.`);
+                log.debug("Exception:", ex);
+                continue;
+            }
+
+            // This must be run for each room.
+            if (!isUserId) {
+                // joined_members surprisingly returns per room profiles, then user profiles in that order
+                // which means we don't need to lookup user's membership info \o/
+                userId = Object.keys(joinedMembers).find((u) => joinedMembers[u].display_name === givenUser);
+                if (!userId) {
+                    log.info(`Failed to find ${givenUser} in ${roomId}`);
+                    continue;
+                }
+                log.debug(`Matching '${givenUser}' as user id`);
+            }
+
+            if (!joinedMembers[userId]) {
+                log.debug(`${userId} was not found in ${roomId}`);
+                continue;
+            } else {
+                foundUser = true;
+            }
+
+            try {
+                await this._main.getBotIntent()[command](roomId, userId, `Requested by ${fromUser.gitterUsername()}`);
+                log.info(`Actioned ${command} against ${userId} in ${roomId}`);
+                actioned = true;
+            } catch (ex) {
+                // TODO: Report these errors to Gitter?
+                log.debug(`Failed to ${command} ${userId} in ${roomId}:`, ex);
+            }
+        }
+
+        if (actioned) {
+            response = null;
```

Comment:
Shouldn't we just return here? We can then also get rid of the below `if` statement.